### PR TITLE
Better test error messages

### PR DESCRIPTION
**Pull request by nkohen**
An initial attempt at improving error messages with testkit

### CODE REVIEW

**Comment by nkohen** [comment]
Example:
```
[info] - should fail with a correct error message *** FAILED ***
[info]   Condition timed out (Thread.java:1559)
[info]   org.scalatest.exceptions.TestFailedException:
[info]   at org.bitcoins.util.AsyncUtil.$anonfun$retryUntilSatisfiedWithCounter$1(AsyncUtil.scala:19)
[info]   at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
[info]   at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
[info]   at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[info]   at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:55)
[info]   at akka.dispatch.BatchingExecutor$BlockableBatch.$anonfun$run$1(BatchingExecutor.scala:91)
[info]   at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[info]   at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
[info]   at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:91)
[info]   at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:40)
[info]   at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(ForkJoinExecutorConfigurator.scala:44)
[info]   at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
[info]   at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
[info]   at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
[info]   at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```
becomes
```
[info] - should fail with a correct error message *** FAILED ***
[info]   Condition timed out (AsyncUtilTest.scala:16)
[info]   org.scalatest.exceptions.TestFailedException:
[info]   at org.bitcoins.util.AsyncUtilTest.$anonfun$new$1(AsyncUtilTest.scala:16)
[info]   at org.scalatest.AsyncFlatSpecLike.transformToOutcomeParam$1(AsyncFlatSpecLike.scala:139)
[info]   at org.scalatest.AsyncFlatSpecLike.$anonfun$registerTestToRun$2(AsyncFlatSpecLike.scala:145)
```

**Comment by nkohen** [comment]
Before trimming a stack trace it  looks like this (in the example above)
```
java.lang.Thread.getStackTrace(Thread.java:1559)
org.bitcoins.util.AsyncUtilTest.$anonfun$new$1(AsyncUtilTest.scala:16)
org.scalatest.AsyncFlatSpecLike.transformToOutcomeParam$1(AsyncFlatSpecLike.scala:139)
org.scalatest.AsyncFlatSpecLike.$anonfun$registerTestToRun$2(AsyncFlatSpecLike.scala:145)
org.scalatest.AsyncTestSuite.$anonfun$transformToOutcome$1(AsyncTestSuite.scala:240)
org.scalatest.AsyncFlatSpecLike$$anon$1.apply(AsyncFlatSpecLike.scala:1698)
org.scalatest.AsyncTestSuite.withFixture(AsyncTestSuite.scala:313)
org.scalatest.AsyncTestSuite.withFixture$(AsyncTestSuite.scala:312)
org.scalatest.AsyncFlatSpec.withFixture(AsyncFlatSpec.scala:2219)
org.scalatest.AsyncFlatSpecLike.invokeWithAsyncFixture$1(AsyncFlatSpecLike.scala:1696)
org.scalatest.AsyncFlatSpecLike.$anonfun$runTest$1(AsyncFlatSpecLike.scala:1709)
org.scalatest.AsyncSuperEngine.runTestImpl(AsyncEngine.scala:292)
org.scalatest.AsyncFlatSpecLike.runTest(AsyncFlatSpecLike.scala:1709)
org.scalatest.AsyncFlatSpecLike.runTest$(AsyncFlatSpecLike.scala:1689)
org.scalatest.AsyncFlatSpec.runTest(AsyncFlatSpec.scala:2219)
org.scalatest.AsyncFlatSpecLike.$anonfun$runTests$1(AsyncFlatSpecLike.scala:1767)
org.scalatest.AsyncSuperEngine.$anonfun$runTestsInBranch$1(AsyncEngine.scala:434)
scala.collection.LinearSeqOptimized.foldLeft(LinearSeqOptimized.scala:126)
scala.collection.LinearSeqOptimized.foldLeft$(LinearSeqOptimized.scala:122)
scala.collection.immutable.List.foldLeft(List.scala:89)
org.scalatest.AsyncSuperEngine.traverseSubNodes$1(AsyncEngine.scala:419)
org.scalatest.AsyncSuperEngine.runTestsInBranch(AsyncEngine.scala:475)
org.scalatest.AsyncSuperEngine.$anonfun$runTestsInBranch$1(AsyncEngine.scala:460)
scala.collection.LinearSeqOptimized.foldLeft(LinearSeqOptimized.scala:126)
scala.collection.LinearSeqOptimized.foldLeft$(LinearSeqOptimized.scala:122)
scala.collection.immutable.List.foldLeft(List.scala:89)
org.scalatest.AsyncSuperEngine.traverseSubNodes$1(AsyncEngine.scala:419)
org.scalatest.AsyncSuperEngine.runTestsInBranch(AsyncEngine.scala:483)
org.scalatest.AsyncSuperEngine.runTestsImpl(AsyncEngine.scala:550)
org.scalatest.AsyncFlatSpecLike.runTests(AsyncFlatSpecLike.scala:1767)
org.scalatest.AsyncFlatSpecLike.runTests$(AsyncFlatSpecLike.scala:1766)
org.scalatest.AsyncFlatSpec.runTests(AsyncFlatSpec.scala:2219)
org.scalatest.Suite.run(Suite.scala:1147)
org.scalatest.Suite.run$(Suite.scala:1129)
org.scalatest.AsyncFlatSpec.org$scalatest$AsyncFlatSpecLike$$super$run(AsyncFlatSpec.scala:2219)
org.scalatest.AsyncFlatSpecLike.$anonfun$run$1(AsyncFlatSpecLike.scala:1812)
org.scalatest.AsyncSuperEngine.runImpl(AsyncEngine.scala:620)
org.scalatest.AsyncFlatSpecLike.run(AsyncFlatSpecLike.scala:1812)
org.scalatest.AsyncFlatSpecLike.run$(AsyncFlatSpecLike.scala:1810)
org.bitcoins.util.AsyncUtilTest.org$scalatest$BeforeAndAfterAll$$super$run(AsyncUtilTest.scala:8)
org.scalatest.BeforeAndAfterAll.liftedTree1$1(BeforeAndAfterAll.scala:213)
org.scalatest.BeforeAndAfterAll.run(BeforeAndAfterAll.scala:210)
org.scalatest.BeforeAndAfterAll.run$(BeforeAndAfterAll.scala:208)
org.bitcoins.util.AsyncUtilTest.run(AsyncUtilTest.scala:8)
org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:314)
org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:507)
sbt.TestRunner.runTest$1(TestFramework.scala:113)
sbt.TestRunner.run(TestFramework.scala:124)
sbt.TestFramework$$anon$2$$anonfun$$lessinit$greater$1.$anonfun$apply$1(TestFramework.scala:282)
sbt.TestFramework$.sbt$TestFramework$$withContextLoader(TestFramework.scala:246)
sbt.TestFramework$$anon$2$$anonfun$$lessinit$greater$1.apply(TestFramework.scala:282)
sbt.TestFramework$$anon$2$$anonfun$$lessinit$greater$1.apply(TestFramework.scala:282)
sbt.TestFunction.apply(TestFramework.scala:294)
sbt.Tests$.$anonfun$toTask$1(Tests.scala:309)
sbt.std.Transform$$anon$3.$anonfun$apply$2(System.scala:46)
sbt.std.Transform$$anon$4.work(System.scala:67)
sbt.Execute.$anonfun$submit$2(Execute.scala:269)
sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:16)
sbt.Execute.work(Execute.scala:278)
sbt.Execute.$anonfun$submit$1(Execute.scala:269)
sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
java.lang.Thread.run(Thread.java:748)
```

File: testkit/src/main/scala/org/bitcoins/util/AsyncUtil.scala [hunk 1]
```diff
@@ -1,5 +1,49 @@
...
+                                      counter,
+                                      maxTries,
+                                      stackTrace)
+
+    AsyncUtil.transformRetryToTestFailure(retryF)(system.dispatcher)
+  }
+}
+
+object AsyncUtil extends AsyncUtil {
+  def transformRetryToTestFailure[T](fut: Future[T])(implicit ec: ExecutionContext): Future[T] = {
```
Comment (Christewart) [inline testkit/src/main/scala/org/bitcoins/util/AsyncUtil.scala:36]: Scaladoc on what this does actually, why we need it?

File: testkit/src/main/scala/org/bitcoins/rpc/RpcUtil.scala [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package org.bitcoins.rpc
+
+import akka.actor.ActorSystem
+import org.bitcoins.util.AsyncUtil
+
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+
+trait RpcUtil extends org.bitcoins.rpc.util.RpcUtil {
```
Comment (Christewart) [inline testkit/src/main/scala/org/bitcoins/rpc/RpcUtil.scala]: abstract class

File: testkit/src/main/scala/org/bitcoins/util/AsyncUtil.scala [hunk 3]
```diff
@@ -1,5 +1,49 @@
...
+import akka.actor.ActorSystem
+import org.scalatest.exceptions.{StackDepthException, TestFailedException}
 
-trait AsyncUtil extends org.bitcoins.rpc.util.AsyncUtil
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration.FiniteDuration
+import scala.util.{Failure, Success}
 
-object AsyncUtil extends AsyncUtil
+trait AsyncUtil extends org.bitcoins.rpc.util.AsyncUtil {
```
Comment (Christewart) [inline testkit/src/main/scala/org/bitcoins/util/AsyncUtil.scala]: abstract class

File: bitcoind-rpc/src/main/scala/org/bitcoins/rpc/util/AsyncUtil.scala [hunk 4]
```diff
@@ -52,28 +52,15 @@ trait AsyncUtil extends BitcoinSLogger {
       message: String,
       caller: Array[StackTraceElement])
       extends Exception(message) {
-    override def printStackTrace(s: PrintStream): Unit = {
-      super.printStackTrace(s)
+    private val relevantStackTrace = caller.tail
```
Comment (Christewart) [inline bitcoind-rpc/src/main/scala/org/bitcoins/rpc/util/AsyncUtil.scala:61]: scaladoc on what this does

**Pull request merged**